src/convertToObject.js

```diff
@@ -6,7 +6,30 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+
+  const cleanedSource = sourceString.trim().replace(/;\s*$/, '');
```

Comment:
[CODE KNOWLEDGE] - There's no need to make a copy before each operation. Methods like `filter` or `map` don't mutate the original array. You can chain the operations directly after the `split` method.